pilot/tools/debug/pilot_cli.go

```diff
@@ -291,24 +312,30 @@ func main() {
 			}
 		}
 	}()
-	pod := NewPodInfo(*proxyTag, resolveKubeConfigPath(*kubeConfig), *proxyType)
+	pod := NewPodInfo(*proxyTag, resolveKubeConfigPath(*kubeConfig), *proxyType, *nodeID)
 
 	var resp *xdsapi.DiscoveryResponse
-	switch *configType {
-	case "lds", "cds":
+	if *resources == "" {
 		resp, err = pod.getXdsResponse(pilot, pod.makeRequest(*configType))
-	case "rds", "eds":
+	} else {
 		resp, err = pod.getXdsResponse(pilot, pod.appendResources(pod.makeRequest(*configType), strings.Split(*resources, ",")))
-	default:
-		log.Errorf("Unknown config type: %q", *configType)
-		os.Exit(1)
 	}
-
 	if err != nil {
 		log.Errorf("Failed to get Xds response for %v. Error: %v", *resources, err)
 		return
 	}
-	strResponse, _ := gogoprotomarshal.ToJSONWithIndent(resp, " ")
+
+	// Fails with "unknown message type "envoy.api.v2.Listener"
+	// Envoy is now using golang/protobuf
```

Comment:
If we are going to start mixing our protos on XDS we should really consider moving ours over as well.. otherwise we have this headache

Reply:
K8S is using gogo. And I don't think they'll move. I would rather keep all the stores consistent.

Unfortunately envoy no longer generates gogo - if we can generate and find a way to use both grpc library ( which has a dep on Envoy APIs with golang proto ) and our envoy API - it would be perfect. 

The gogo protos are clearly cleaner, nicer, faster - we only use XDS because envoy changed.